process_report/process_report.py

```diff
@@ -112,21 +194,107 @@ def remove_billables(dataframe, pi, projects, output_file):
+def export_billables(dataframe, output_file):
+    dataframe.to_csv(output_file, index=False)
+
+
 def export_pi_billables(dataframe: pandas.DataFrame, output_folder):
     if not os.path.exists(output_folder):
         os.mkdir(output_folder)
 
-    invoice_month = dataframe['Invoice Month'].iat[0]
-    pi_list = dataframe['Manager (PI)'].unique()
+    invoice_month = dataframe[INVOICE_DATE_FIELD].iat[0]
+    pi_list = dataframe[PI_FIELD].unique()
 
     for pi in pi_list:
-        pi_projects = dataframe[dataframe['Manager (PI)'] == pi]
-        pi_instituition = pi_projects['Institution'].iat[0]
+        if pandas.isna(pi): 
+            continue
+        pi_projects = dataframe[dataframe[PI_FIELD] == pi]
+        pi_instituition = pi_projects[INSTITUTION_FIELD].iat[0]
         pi_projects.to_csv(output_folder + f"/{pi_instituition}_{pi}_{invoice_month}.csv")
-        
+
+
+def apply_credits_new_pi(dataframe, old_pi_file):
+    new_pi_credit_code = "0002"
+    new_pi_credit_amount = 1000
+
+    dataframe[CREDIT_FIELD] = None
+    dataframe[CREDIT_CODE_FIELD] = None
+    dataframe[BALANCE_FIELD] = 0
+
+    old_pi_dict = load_old_pis(old_pi_file)
+
+    current_pi_list = dataframe[PI_FIELD].unique()
+    invoice_month = dataframe[INVOICE_DATE_FIELD].iat[0]
+
+    for pi in current_pi_list:
+        pi_projects = dataframe[dataframe[PI_FIELD] == pi]
+
+        if is_old_pi(old_pi_dict, pi, invoice_month):
+            for i, row in pi_projects.iterrows():
+                dataframe.at[i, BALANCE_FIELD] = row[COST_FIELD]
+        else:
+            remaining_credit = new_pi_credit_amount
+            for i, row in pi_projects.iterrows():
+                project_cost = row[COST_FIELD]
+                applied_credit = min(project_cost, remaining_credit)
+
+                dataframe.at[i, CREDIT_FIELD] =  applied_credit
+                dataframe.at[i, CREDIT_CODE_FIELD] = new_pi_credit_code
+                dataframe.at[i, BALANCE_FIELD] = row[COST_FIELD] - applied_credit
+                remaining_credit -= applied_credit
+
+                if remaining_credit == 0:
+                    break
+    
+    return dataframe
+
+
+def add_institution(dataframe: pandas.DataFrame):
+    """Determine every PI's institution name, logging any PI whose institution cannot be determined
+    This is performed by `get_institution_from_pi()`, which tries to match the PI's username to 
+    a list of known institution email domains (i.e bu.edu), or to several edge cases (i.e rudolph) if
+    the username is not an email address.
+    
+    Exact matches are then mapped to the corresponding institution name. 
+
+    I.e "foo@bu.edu" would match with "bu.edu", which maps to the instition name "Boston University"
+
+    The list of mappings are defined in `institute_map.json`.
+    """
+    for i, row in dataframe.iterrows():
+        pi_name = row[PI_FIELD]
+        if pandas.isna(pi_name): 
+            print(f"Project {row[PROJECT_FIELD]} has no PI") # Nan check
```

Comment:
This comment is unnecessary because the function name describes the same thing due to its name `isna`.